src/next/scval_validations.rs

```diff
@@ -10,11 +10,11 @@ pub trait Validate {
 impl Validate for ScVal {
     type Error = Error;
 
-    fn validate(&self) -> Result<(), Self::Error> {
+    fn validate(&self) -> Result<(), <Self as Validate>::Error> {
```

Comment:
Same question here.

Reply:
same here. any changes to returns that used Self::Error anywhere in the code are disambiguation (most of the SDK changes look like that too)